cmd/bootstrap/transit/main.go

```diff
@@ -458,3 +476,75 @@ func fileExists(filename string) bool {
 	}
 	return !info.IsDir()
 }
+
+// moveFile moves a file from source to destination where src and dst are full paths including the filename
+func moveFile(src, dst string) error {
+
+	// check if source file exist
+	if !fileExists(src) {
+		return fmt.Errorf("file not found: %s", src)
+	}
+
+	// create the destination dir if it does not exist
+	destinationDir := filepath.Dir(dst)
+	err := os.MkdirAll(destinationDir, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create directory %s: %w", destinationDir, err)
+	}
+
+	// create the destination file
+	destination, err := os.Create(dst)
+	if err != nil {
+		return fmt.Errorf("failed to create file %s: %w", dst, err)
+	}
+	defer destination.Close()
+
+	// open the source file
+	source, err := os.Open(src)
+	if err != nil {
+		return fmt.Errorf("failed to open file %s: %w", src, err)
+	}
+
+	// copy the source file to the destination file
```

Comment:
Why not using [Renaming](https://golang.org/pkg/os/#Rename), would be much faster

Reply:
So I came across this thread (https://stackoverflow.com/questions/50740902/move-a-file-to-a-different-drive-with-go) which talks about problems with `renaming` files if the destination is a directory on a different disk. Hence, to take the most safest approach I did a `copy, remove` as suggested over there.

Reply:
Could we try renaming first and if it hits the error "Different disk" error, then do real copy?

I think in most cases, the root.checkpoint file will be downloaded to the same disk, no?

But this suggestion is optional.

Reply:
updated